lib/Driver/Driver.cpp

```diff
@@ -2031,6 +2031,12 @@ const ToolChain *Driver::getToolChain(const ArgList &Args) const {
       TC = new toolchains::Darwin(*this, Target);
       break;
     case llvm::Triple::Linux:
+      if (Target.isAndroid()) {
```

Comment:
I am also agree renaming the toolchain ```Cygwin```. I thought rename it if I add toolchain  ```MSVC``` or ```Mingw``` someday.
and...
How about renaming ```GenericUnix``` to ```GNUToolchain```?
The heirarchy will be 

```
GNUToolchain
    Android
    Cygwin
    Mingw
```
Cygwin uses the GNU toolchain and has Unix compatible libraries, but it is not UNIX OS.